packages/network/src/utils.ts

```diff
@@ -8,15 +9,27 @@ export function messagePayload(msg: Message) {
   return `(${msg.version},${msg.id},${keccak256(msg.data)},${msg.timestamp})`;
 }
 
-// Remove zero padding from all entity ids
-export function formatEntityID(entityID: string | EntityID | BigNumber): EntityID {
-  if (BigNumber.isBigNumber(entityID) || entityID.substring(0, 2) === "0x") {
+// For v1 entities (BigNumber or hex strings) and single-key v2 entities (hex strings), strip zero padding
+// For composite-key v2 entities (hex strings concatenated with `:`), leave the ID as is
+// See `keyTupleToEntityID`
+export function normalizeEntityID(entityID: string | EntityID | BigNumber): EntityID {
+  if (BigNumber.isBigNumber(entityID)) {
```

Comment:
does this work if the entity id is concatenated with `:` or does ethers still think it's a bignumber but then error on `toHexString()`? (Asking bc I thought we had an issue with that yesterday)

Reply:
the issue came not from ethers, but our bad logic that assumed "string starts with 0x can be converted to BigNumber"

Reply:
here's what `BigNumber.isBigNumber` looks like: <img width="500" alt="image" src="https://user-images.githubusercontent.com/508855/225686579-e60974c7-6f3d-4cee-a29d-6f992d777ea9.png">